### PR TITLE
Roi image

### PR DESCRIPTION
**Pull request by bnmajor**
Add support for requesting an NgffImage of the current ROI

### CODE REVIEW

File: itkwidgets/viewer.py [hunk 1]
```diff
@@ -323,6 +344,34 @@ async def get_current_scale(self):
...
+            multiscales = from_ngff_zarr(store)
+            loaded_image = multiscales.images[scale]
+            roi_data = loaded_image.data[roi_slices]
+            return to_ngff_image(
+                roi_data,
+                dims=loaded_image.dims,
+                scale=loaded_image.scale,
+                translation=roi_region[0]
+            )
+        raise Exception(f'No image data found for {name}.')
```
Comment (thewtex) [inline itkwidgets/viewer.py:373]: ValueError

File: itkwidgets/viewer.py [hunk 2]
```diff
@@ -323,6 +344,34 @@ async def get_current_scale(self):
...
+        roi_region = await self.get_roi_region()
+        if store := self.stores.get(name):
+            multiscales = from_ngff_zarr(store)
+            loaded_image = multiscales.images[scale]
+            roi_data = loaded_image.data[roi_slices]
+            return to_ngff_image(
+                roi_data,
+                dims=loaded_image.dims,
+                scale=loaded_image.scale,
+                translation=roi_region[0]
```
Comment (thewtex) [inline itkwidgets/viewer.py:371]: please also pass other kwargs, e.g. units

File: itkwidgets/viewer.py [hunk 3]
```diff
@@ -408,7 +458,21 @@ def set_label_image(self, label_image: Image):
             self.viewer_rpc.itk_viewer.setPointSets(label_image)
     @fetch_value
     async def get_label_image(self):
```
Comment (thewtex) [inline itkwidgets/viewer.py:460]: return type annotation

**Review by thewtex — CHANGES_REQUESTED** [review]
bueno! a few items inline

File: itkwidgets/viewer.py (hunk 1 above)
Comment (bnmajor) [reply itkwidgets/viewer.py]: Fixed

File: itkwidgets/viewer.py (hunk 2 above)
Comment (bnmajor) [reply itkwidgets/viewer.py]: Added in

**Comment by bnmajor** [comment]
<p>Fixed</p>

---

 <a href='https://app.reviewnb.com/InsightSoftwareConsortium/itkwidgets/pull/710/discussion/#comment-1856032131'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by bnmajor** [comment]
<p>Removed</p>

---

 <a href='https://app.reviewnb.com/InsightSoftwareConsortium/itkwidgets/pull/710/discussion/#comment-1856032156'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by bnmajor** [comment]
<p>Moved</p>

---

 <a href='https://app.reviewnb.com/InsightSoftwareConsortium/itkwidgets/pull/710/discussion/#comment-1856032215'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by bnmajor** [comment]
> View / edit / reply to [this conversation](https://app.reviewnb.com/InsightSoftwareConsortium/itkwidgets/pull/710/discussion/#comment-1856032280) on ReviewNB
> 
> _thewtex commented on 2023-12-14T15:13:09Z_ ----------------------------------------------------------------
> 
> Line #7.    viewer3 = view(roi_image, rotate=True)
> this cell and the one above can be removed

Just to clarifty: we don't want the `viewer3` example at all anymore, correct?

**Review by thewtex — APPROVED** [review]

**Pull request merged**